app/controllers/auth_services_controller.rb

```diff
@@ -1,4 +1,12 @@
 class AuthServicesController < ApplicationController
+  def new
+    referer_path = URI(request.referer).path
+    if Rails.application.routes.recognize_path(referer_path)[:controller].in?(%w(workshops events courses meetings))
+      session[:referer_path] = referer_path
+    end
+    redirect_to "/auth/github"
+  end
+
```

Comment:
This is for the redirect functionality. I created this method to use for the '/login' path. The login link on the WECMs pages will go here instead of immediately to the github auth so we can set a session cookie.